core/common/src/AtmosphericScattering.ts

```diff
@@ -0,0 +1,173 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module DisplayStyles
+ */
+
+import { Point3d, XYZProps } from "@itwin/core-geometry";
+import { ColorDef, ColorDefProps } from "./ColorDef";
+
+export const defaultAtmosphericScatteringProps: Required<AtmosphericScatteringProps> =
+  {
+    earthCenter: { x: 0.0, y: 0.0, z: -6_190_000.0 },
+    atmosphereRadius: 6_290_100.0,
+    earthRadius: 6_190_000.0,
+    densityFalloff: 5.0,
+    scatteringStrength: 0.01,
+    wavelenghts: [700.0, 530.0, 440.0],
+    numInScatteringPoints: 10,
+    numOpticalDepthPoints: 10,
+    isPlanar: false,
+    // earthCenter: {x: 1000.0, y: -10000.0, z: -105000.0},
+    // atmosphereRadius: 110000.0,
+    // earthRadius: 9000.0,
+    // densityFalloff: 5.0,
+    // scatteringStrength: 1.0,
+    // wavelenghts: [700.0, 530.0, 440.0],
+    // numInScatteringPoints: 10,
+    // numOpticalDepthPoints: 10,
+    // isPlanar: false,
+  };
+
+/**
+ * @public
+ */
+export interface AtmosphericScatteringProps {
+  earthCenter?: XYZProps;
+  atmosphereRadius?: number;
+  earthRadius?: number;
+  densityFalloff?: number;
+  scatteringStrength?: number;
+  wavelenghts?: number[];
+  numInScatteringPoints?: number;
+  numOpticalDepthPoints?: number;
+  isPlanar?: boolean;
+}
+
+/**
+ * @public
+ */
+export class AtmosphericScattering {
+  public readonly earthCenter: Point3d;
+  public readonly atmosphereRadius: number;
+  public readonly earthRadius: number;
+  public readonly densityFalloff: number;
+  public readonly scatteringStrength: number;
+  public readonly wavelenghts: number[];
+  public readonly numInScatteringPoints: number;
+  public readonly numOpticalDepthPoints: number;
+  public readonly isPlanar: boolean;
+
+  public equals(other: AtmosphericScattering): boolean {
+    if (this.earthCenter !== other.earthCenter) return false;
+    if (this.atmosphereRadius !== other.atmosphereRadius) return false;
+    if (this.earthCenter !== other.earthCenter) return false;
+    if (this.densityFalloff !== other.densityFalloff) return false;
+    if (this.scatteringStrength !== other.scatteringStrength) return false;
+    if (this.wavelenghts !== other.wavelenghts) return false;
+    if (this.numInScatteringPoints !== other.numInScatteringPoints)
+      return false;
+    if (this.numOpticalDepthPoints !== other.numOpticalDepthPoints)
+      return false;
+    if (this.isPlanar !== other.isPlanar) return false;
+    return true;
+  }
+
+  private constructor(json?: AtmosphericScatteringProps) {
+    if (json === undefined) {
```

Comment:
You can shrink and simplify this constructor considerably by using spread operator and/or null coalescence operator.
e.g.,:
`this.atmosphereRadius = json?.atmosphereRadius ?? defaultAtmosphericScatteringProps.atmosphereRadius;`
or
`json = { ...defaultAtmosphericScatteringProps, ...json };`
However be careful with the latter as `...json` may make some properties `undefined` if caller explicitly did something like `{ atmosphereRadius: undefined }`.